pyne/xs/data_source.py

```diff
@@ -213,9 +214,29 @@ def discretize(self, nuc, rx, temp=300.0, src_phi_g=None, dst_phi_g=None):
         src_sigma = self.reaction(nuc, rx, temp)
         dst_sigma = None if src_sigma is None else group_collapse(src_sigma,
                                                         src_phi_g, dst_phi_g,
-                                                        self._src_to_dst_matrix)
+                                                        self._src_to_dst_matrix, wgts=self.slf_shld_wgts[nuc])
         return dst_sigma
 
+    def shield_weights(self, mat, temp):
+        """Builds the weights used during the self shielding calculations. 
+        Parameters
+        ----------
+        mat : array of floats.  
+            A map of the number densities of each of the nuclides of the material for 
+            which self-shielding is being calculated. 
+        data_source: pyne data_source
+            Contains the cross section information for the isotopes in the material 
+            that is experiencing the self shielding. 
+        """
+        weight = {}
+        for iso in mat:
```

Comment:
Use nuc instead of iso. See the style guide: http://pyne.io/devsguide/style_guide.html#terminology